ignite/engine/__init__.py

```diff
@@ -91,20 +101,54 @@ def create_supervised_trainer(
 
     device_type = device.type if isinstance(device, torch.device) else device
     on_tpu = "xla" in device_type if device_type is not None else False
+    has_native_amp, has_apex_amp = False, False
 
     if on_tpu and not idist.has_xla_support:
         raise RuntimeError("In order to run on TPU, please install PyTorch XLA")
 
+    if amp and LooseVersion(torch.__version__) >= LooseVersion("1.6.0"):
+        from torch.cuda.amp import autocast
+
+        has_native_amp = True
```

Comment:
As for the scaler, we can create a default one if user do not provide one in the args.

Reply:
For the scaler, I let users create by themselves if they want to save `state_dict` since engine instance is the only return.

Reply:
Thanks, that's a great point to think about ! We can add `apex.amp` or `scaler` to `engine.state` as attributes and also add to `state_dict_user_keys` to checkpoint if trainer is checkpointed: https://pytorch.org/ignite/engine.html#ignite.engine.engine.Engine.state_dict

Reply:
I see! Great
So should we still have `scaler` arg?

Reply:
Good question! Maybe, it is unnecessary, let's see...

Reply:
For now, I will push current progress as I want to run GPU test on colab.

Reply:
I think I can enable GPU tests for PRs now as now we can filter cancel workflow if code was not updated...

Reply:
@vfdev-5 we can't create scaler internally. `GradScaler` is used for gradients that are near to zero after autocast.
If the gradients are not near zero, we don't need that. It's user responsibility to pass scaler depending on their model's gradient.
https://pytorch.org/tutorials/recipes/recipes/amp_recipe.html#adding-autocast

And also, if `amp` and `scaler` are passed but no `grad_norm_kwargs`, `scaler.step()` will do `unscale_()`
so it's ok to call only `unscale_()` if `grad_norm_kwargs` are passed.

Reply:
I think what we could do is the following. We follow the guidelines from amp recipe "All together: Automatic Mixed Precision" no matter values of the gradient. GradScaler is created internally as discussed above and we keep scaler arg which could be interpreted as following: 
- None -> create internal one, 
- instance of GradScaler -> use provided one
- False -> disable internal one

```python
def create_supervised_trainer(
    model: torch.nn.Module,
    optimizer: torch.optim.Optimizer,
    prepare_batch: Callable = _prepare_batch,
    output_transform: Callable = lambda x, y, y_pred, loss: loss.item(),
    deterministic: bool = False,
    amp: bool = False,
    scaler: Optional[Union[bool, "torch.cuda.amp.GradScaler"]] = None,  # type: ignore[assignment]
    **grad_norm_kwargs: Any,
) -> Engine:
```
What do you think ?

Reply:
> And also, if amp and scaler are passed but no grad_norm_kwargs, scaler.step() will do unscale_()
so it's ok to call only unscale_() if grad_norm_kwargs are passed.

This is what we do now, right ?

Reply:
> - None -> create internal one,
> - instance of GradScaler -> use provided one
> - False -> disable internal one

I think `bool` and instance of GradScaler is enough.
- True -> create internal one,
- instance of GradScaler -> use provided one
- False -> disable internal one